cpp/oneapi/dal/algo/covariance/backend/cpu/finalize_compute_kernel_dense.cpp

```diff
@@ -57,7 +57,20 @@ static compute_result<Task> call_daal_kernel_finalize(const context_cpu& ctx,
     const auto daal_nobs_matrix = interop::convert_to_daal_table<Float>(input.get_partial_n_rows());
 
     auto result = compute_result<Task>{}.set_result_options(desc.get_result_options());
-
+    auto rows_count_global =
+        row_accessor<const Float>(input.get_partial_n_rows()).pull({ 0, -1 })[0];
+    daal_covariance::internal::Hyperparameter daal_hyperparameter;
+    /// the logic of block size calculation is copied from DAAL,
+    /// to be changed to passing the values from the performance model
+    std::int64_t blockSize = 140;
```

Comment:
why 140? and would it be 140 when row_count > 50000?

Reply:
I've taken the `blockSize` from batch kernel. May be @Vika-F can clarify this value?

Reply:
This logic was copied from DAAL. The constants were defined in DAAL as the result of a series of performance measurements performed in the past. In DAAL, batch and online implementations share large amount of code, including those constants which are actually performance related hyperparameters of the algorithm.  
In the future this logic will be moved into `hyperparameter` classes, this was already implemented in batch oneDAL algorithm, and will be replaced further by more intelligent solution (not just hard-coded constants).

Reply:
@Vika-F thank you for clarification, seems reasonable for now. Based on this logic block_size would be 140 when number of rows is > 50k, is that intended as well? Seems a bit odd to set to 1024 for larger data but then back to 140 for very large data.

Reply:
It might be that in case of large number of rows the particular size of the block is not that sufficient - as there is enough work available for each thread. But this is only my guess. The original data from the performance experiments based on which those constants were defined is already unavailable.
And that's why we are moving towards more intelligent solution than hard-coded constants.